packages/grafana-data/src/field/fieldOverrides.test.ts

```diff
@@ -353,6 +362,27 @@ describe('setDynamicConfigValue', () => {
     expect(config.custom.property1).toEqual('applied');
   });
 
+  it('applies overrides even when shouldApply returns false', () => {
```

Comment:
I see what we are doing but it feels like we probably should rename the `shouldApply` to something else or change so it can make the decision and return `true` in this scenario.

Reply:
Will give it some thought and see if I can give some better feedback.